x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/predicate/operator/comparison/EqualsTests.java

```diff
@@ -217,8 +217,8 @@ public static Iterable<Object[]> parameters() {
     }
 
     private static String typeErrorString =
-        "boolean, cartesian_point, cartesian_shape, datetime, date_nanos, double, geo_point, geo_shape, integer, ip, keyword, long, text, "
```

Comment:
for the comparison operators tests I did not need to make any other changes because they already use `TestCase.stringCases`:

https://github.com/elastic/elasticsearch/blob/f32051f4629f6f1a4192a615ec0cb5e294089fb2/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/predicate/operator/comparison/EqualsTests.java#L147-L154

The same applies for the rest of the comparison predicates:

https://github.com/elastic/elasticsearch/blob/f32051f4629f6f1a4192a615ec0cb5e294089fb2/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/predicate/operator/comparison/GreaterThanOrEqualTests.java#L137-L144

https://github.com/elastic/elasticsearch/blob/f32051f4629f6f1a4192a615ec0cb5e294089fb2/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/predicate/operator/comparison/GreaterThanTests.java#L137-L144

https://github.com/elastic/elasticsearch/blob/f32051f4629f6f1a4192a615ec0cb5e294089fb2/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/predicate/operator/comparison/LessThanOrEqualTests.java#L137-L144

https://github.com/elastic/elasticsearch/blob/f32051f4629f6f1a4192a615ec0cb5e294089fb2/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/predicate/operator/comparison/LessThanTests.java#L137-L144

https://github.com/elastic/elasticsearch/blob/f32051f4629f6f1a4192a615ec0cb5e294089fb2/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/predicate/operator/comparison/NotEqualsTests.java#L145-L151

`TestCaseSupplier` uses `DataType.stringTypes()` which contains `semantic_text`:

https://github.com/elastic/elasticsearch/blob/f32051f4629f6f1a4192a615ec0cb5e294089fb2/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/TestCaseSupplier.java#L84-L95